Index: xpdf/GfxState.cc
===================================================================
--- xpdf/GfxState.cc	(rev 0)
+++ xpdf/GfxState.cc	(working copy)
@@ -4731,7 +4731,10 @@ double GfxState::getTransformedFontSize() {
   y1 = textMat[3] * fontSize;
   x2 = ctm[0] * x1 + ctm[2] * y1;
   y2 = ctm[1] * x1 + ctm[3] * y1;
-  return sqrt(x2 * x2 + y2 * y2);
+  double res = sqrt(x2 * x2 + y2 * y2);
+  // Font size above threshold might cause overflow
+  // when converted into integral coordinates
+  return fmin(res, 1e6);
 }
 
 void GfxState::getFontTransMat(double *m11, double *m12,
